services:
  mysql:
    image: mysql:8.0-debian
    container_name: pythonlab.mysql
    environment:
      MYSQL_ROOT_PASSWORD: '${MYSQL_ROOT_PASSWORD}'
      MYSQL_DATABASE: '${MYSQL_DATABASE}'
      MYSQL_USER: '${MYSQL_USER}'
      MYSQL_PASSWORD: '${MYSQL_PASSWORD}'
    ports:
      - "3306:3306"
    healthcheck:
      test: ["CMD-SHELL", "mysqladmin ping -h localhost -u root --password=root"]
      interval: 5s
      timeout: 5s
      retries: 5
    volumes:
      - ./db-data:/var/lib/mysql

  phpmyadmin:
    image: phpmyadmin:5.2.1
    container_name: pythonlab.phpmyadmin
    ports:
      - "8082:80"
    environment:
      PMA_HOST: mysql
      PMA_USER: '${MYSQL_USER}'
      PMA_PASSWORD: '${MYSQL_PASSWORD}'
      PMA_ARBITRARY: 1
    depends_on:
      mysql:
        condition: service_healthy         

networks:
  cupom-fiscal-net:
    driver: bridge